# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/arm/Lowering-arm.h
# Commit: aad37f6cbf03
# Full Hash: aad37f6cbf034f5ec3e0b04989aa025c7c899697
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 15:10:42
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6.2: Establish dependency from tls for x86 callWithABI div/mod i64. r=lth
#   
#   x86 has few register so to do div/mod for i64 it call the runtime
#   and clobber almost all gp registers including WasmTlsReg.
#   To be able to call c++ runtime via Builtin thunk we need to set up WasmTlsReg.
# ==============================================================================

diff -r 05007ff436a5 -r aad37f6cbf03 js/src/jit/arm/Lowering-arm.h
--- a/js/src/jit/arm/Lowering-arm.h	Fri Sep 11 07:15:21 2020 +0000
+++ b/js/src/jit/arm/Lowering-arm.h	Fri Sep 11 08:05:01 2020 +0000
@@ -72,7 +72,9 @@
   void lowerDivI(MDiv* div);
   void lowerModI(MMod* mod);
   void lowerDivI64(MDiv* div);
+  void lowerWasmBuiltinDivI64(MWasmBuiltinDivI64* div);
   void lowerModI64(MMod* mod);
+  void lowerWasmBuiltinModI64(MWasmBuiltinModI64* mod);
   void lowerUDivI64(MDiv* div);
   void lowerUModI64(MMod* mod);
   void lowerMulI(MMul* mul, MDefinition* lhs, MDefinition* rhs);